src/shell/shell.rs

```diff
@@ -11,23 +13,38 @@ pub trait Shell: Debug {
     fn to_clap_shell(&self) -> clap_complete::Shell;
 }
 
-#[cfg(windows)]
-pub const AVAILABLE_SHELLS: &[&str; 5] = &["cmd", "powershell", "bash", "zsh", "fish"];
-
-#[cfg(unix)]
-pub const AVAILABLE_SHELLS: &[&str; 4] = &["bash", "zsh", "fish", "powershell"];
+#[derive(Debug, Clone, ValueEnum)]
+pub enum Shells {
+    Bash,
+    Zsh,
+    Fish,
+    PowerShell,
+    #[cfg(windows)]
+    Cmd,
+}
 
-impl std::str::FromStr for Box<dyn Shell> {
-    type Err = String;
+impl Display for Shells {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Shells::Bash => write!(f, "bash"),
+            Shells::Zsh => write!(f, "zsh"),
+            Shells::Fish => write!(f, "fish"),
+            Shells::PowerShell => write!(f, "powershell"),
+            #[cfg(windows)]
+            Shells::Cmd => write!(f, "cmd"),
+        }
+    }
+}
 
-    fn from_str(s: &str) -> Result<Box<dyn Shell>, Self::Err> {
-        match s {
-            "cmd" => Ok(Box::from(super::windows_cmd::WindowsCmd)),
-            "zsh" => Ok(Box::from(super::zsh::Zsh)),
-            "bash" => Ok(Box::from(super::bash::Bash)),
-            "fish" => Ok(Box::from(super::fish::Fish)),
-            "powershell" => Ok(Box::from(super::powershell::PowerShell)),
-            shell_type => Err(format!("I don't know the shell type of {shell_type:?}",)),
+impl From<Shells> for Box<dyn Shell> {
```

Comment:
This is a first good step, in a later PR we can remove `Box<dyn Shell>` entirely by implementing `Shell` for `Shells` and forward the calls

Reply:
So kinda like this?
https://github.com/Schniz/fnm/pull/991/commits/7781b5d87354c1f9500ecfafde1aba43d6dd5e37